### PR TITLE
gulp task package refactored, small fixes in other tasks

### PR DESCRIPTION
**Pull request by ppitonak**
* the most important Gulp task is split into multiple tasks so it is easier to understand and debug/improve
* return and callbacks added where necessary for tasks to run correctly
* known issue: Gulp doesn't log tasks ideally

### CODE REVIEW

**Comment by ppitonak** [comment]
@dgolovin, @jrichter1 please review and merge if everything is OK.

**Comment by jrichter1** [comment]
The 'dist' script creates a simple installer and then overwrites it with a bundled installer - both files have the same name.

File: gulpfile.js [hunk 1]
```diff
@@ -38,6 +39,9 @@ let zaSfxExe = path.join(zaRoot, '7zS.exe');
 let zaExtra7z = path.join(zaRoot, '7z920_extra.7z');
 let rhZip = path.join(zaRoot, 'resource_hacker.zip');
 let rhExe = path.join(zaRoot, 'ResourceHacker.exe');
+let zaElectronPackage = path.join(zaRoot, artifactName + '-win32-x64');
+let bundled7z = path.join(zaRoot, artifactName +'-win32-x64.7z');
+let installerExe = path.join(zaRoot, artifactName + '-' + pjson.version + artifactType + '-installer.exe');
```
Comment (jrichter1) [inline gulpfile.js]: Declaring installerExe here means that artifactType will always be empty, hence both simple and bundled installers will have the same name.

**Comment by ppitonak** [comment]
@jrichter1 fixed

**Comment by jrichter1** [comment]
Released to master.

**Pull request closed without merging**